backend/server/api/services/todos.service.js

```diff
@@ -18,6 +18,17 @@ export class ToDoService {
     const newTodo = allTodos.child(newTodoRef.key).toJSON();
     return { ...newTodo, key: newTodoRef.key };
   }
+
+  async all() {
+    const allTodos = await this.db.ref("todo").once("value");
```

Comment:
You have to convert this firebase data to a Javascript array.
You can do `allTodos.forEach(...)` and push to an array that you'll return.

Reply:
Done. Also the function that test this now checks for an array instead of an object